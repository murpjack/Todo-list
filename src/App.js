import React from "react";
import { Provider } from './store';

import TodoList from "./components/TodoList";
import Input from "./components/Input";

export default function App() {
  return (
    <Provider>
      <div className="todo">
        <header className="App-header">WHAT TODO</header>
        <TodoList />
        <Input />
      </div>
    </Provider>
  );
}
