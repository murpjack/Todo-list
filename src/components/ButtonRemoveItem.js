import React, { useContext } from "react";
import { OrganiserContext } from "../context";
import PropTypes from "prop-types";
import Future from "fluture/index.js";
import { removeItem } from "../actions/index";

export const ButtonRemoveItem = ({ currentIdx }) => {
  const { state, dispatch } = useContext(OrganiserContext);
  const { list } = state;

  function removeItemClick() {
    const updatedList = list.filter((item, idx) => {
      if (idx !== currentIdx) return item;
      return false;
    });

    Future.of(updatedList)
      .map(removeItem)
      .value(dispatch);
  }

  return (
    <button
      className="item__button item__button--remove"
      onClick={removeItemClick}
    >
      x
    </button>
  );
};
ButtonRemoveItem.propTypes = {
  currentIdx: PropTypes.number.isRequired,
  current: PropTypes.object.isRequired
};

export default ButtonRemoveItem;
