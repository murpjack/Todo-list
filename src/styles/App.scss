@import "./variables";
@import "./todo-list";

.App {
  background-color: $color-bg;
  box-sizing: border-box;
  color: $color-font;
  margin: 0 auto;
  min-height: 100vh;
  padding: 30px 0;
  text-align: center;
  width: 100%;
}

.App-header {
  font-size: calc(15px + 2vmin);
  font-weight: 600;
  letter-spacing: 2px;
  margin: 0 0 30px;
text-align: center;
}
