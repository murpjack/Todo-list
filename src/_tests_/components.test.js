import React, {
  useContext
} from "react";

import {
  shallow
} from "enzyme";

import {
  OrganiserContext
} from "../context";

import {
  addItem,
  removeItem,
  editItem,
  toggleItemComplete
} from "../actions/index";

import renderer from 'react-test-renderer';

import App from "../App";
import ButtonRemoveItem from "../components/ButtonRemoveItem";

it("renders", () => {
  shallow(<App />);
});

describe("App", () => {
  it("renders successfully", () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("ButtonRemoveItem", () => {
      it("Test click event", () => {
          const mockCallBack = jest.fn();
          const item = {
            id: 1,
            name: "Write Todo list",
            isComplete: false
          };
          const button = shallow( <ButtonRemoveItem current={ item } />);
            button.find("button").simulate("click");
            // expect(dispatch.mock.calls.length).toEqual(1);
            console.log(1, mockCallBack)
            console.log(2, mockCallBack.mock.calls)
            expect(mockCallBack.mock.calls.length).toEqual(1);
          });
      });
