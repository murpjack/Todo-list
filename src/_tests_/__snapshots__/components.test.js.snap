// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders successfully 1`] = `
<div
  className="todo"
>
  <header
    className="App-header"
  >
    WHAT TODO
  </header>
  <ul
    className="todo__list"
  >
    <li
      className="todo__item item"
    >
      <input
        checked={true}
        className="item__toggle item__toggle--complete"
        onChange={[Function]}
        type="checkbox"
      />
      <input
        className="item__name"
        defaultValue="Write TODO list"
        id="inputTodo"
        onKeyUp={[Function]}
        type="text"
      />
      <button
        className="item__button item__button--remove"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </li>
    <li
      className="todo__item item"
    >
      <input
        checked={false}
        className="item__toggle "
        onChange={[Function]}
        type="checkbox"
      />
      <input
        className="item__name"
        defaultValue="看电影"
        id="inputTodo"
        onKeyUp={[Function]}
        type="text"
      />
      <button
        className="item__button item__button--remove"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </li>
    <li
      className="todo__item item"
    >
      <input
        checked={false}
        className="item__toggle "
        onChange={[Function]}
        type="checkbox"
      />
      <input
        className="item__name"
        defaultValue="Make something"
        id="inputTodo"
        onKeyUp={[Function]}
        type="text"
      />
      <button
        className="item__button item__button--remove"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </li>
  </ul>
  <div>
    <input
      className="todo__input"
      id="inputTodo"
      maxLength="70"
      onKeyDown={[Function]}
      placeholder="Add another task"
      type="text"
    />
     
  </div>
</div>
`;
